bot/exts/help_channels/_cog.py

```diff
@@ -113,33 +129,48 @@ async def rename_help_post(self, ctx: commands.Context, *, title: str) -> None:
 
         await ctx.channel.edit(name=title)
 
-    @commands.Cog.listener()
-    async def on_thread_create(self, thread: discord.Thread) -> None:
+    @commands.Cog.listener("on_message")
+    async def new_post_listener(self, message: discord.Message) -> None:
         """Defer application of new post logic for posts the help forum to the _channel helper."""
-        if thread.parent_id != self.help_forum_channel_id:
+        if not isinstance(message.channel, discord.Thread):
+            return
+        thread = message.channel
+
+        if not message.id == thread.id:
+            # Opener messages have the same ID as the thread
+            return
+
+        if thread.parent_id != self.help_forum_channel.id:
             return
 
         await self.post_with_disallowed_title_check(thread)
-        await _channel.help_thread_opened(thread)
+        await _channel.help_post_opened(thread)
+
+        delay = min(constants.HelpChannels.deleted_idle_minutes, constants.HelpChannels.idle_minutes) * 60
+        self.scheduler.schedule_later(
+            delay,
+            thread.id,
+            _channel.maybe_archive_idle_post(thread, self.scheduler)
```

Comment:
I think we should set `has_task=False` here, since there won't be an event in the `scheduler` for a new post and so `scheduler.cancel` will raise a `ValueError`

Reply:
The code you just quoted is adding this coro as a task to the scheduler

Reply:
The scheduler will have a task for the post - the one scheduled in these lines.

Reply:
Ohh, right. Gotcha.